R/pkg/R/sparkR.R

```diff
@@ -376,6 +377,12 @@ sparkR.session <- function(
     overrideEnvs(sparkConfigMap, paramMap)
   }
 
+  # NOTE(shivaram): Set default warehouse dir to tmpdir to meet CRAN requirements
+  # See SPARK-18817 for more details
+  if (!exists("spark.sql.default.warehouse.dir", envir = sparkConfigMap)) {
```

Comment:
After rethinking it, we might not need to add an extra sql conf. We just need to know whether the value of `spark.sql.warehouse.dir` is from the users or the original default. If it is the default, R can simply change it. 

Maybe it is a good to-have feature for users to know whether the SQLConf value is from users or from the default. cc @cloud-fan